test/jdk/java/awt/Graphics2D/CopyAreaOOB.java

```diff
@@ -26,17 +26,73 @@
  * @key headful
  * @bug 6430601 8198613
  * @summary Verifies that copyArea() works properly when the
- * destination parameters are outside the destination bounds.
+ *          destination parameters are outside the destination bounds.
  * @run main/othervm CopyAreaOOB
- * @author campbelc
  */
 
-import java.awt.*;
-import java.awt.image.*;
+import java.awt.Canvas;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.GraphicsConfiguration;
+import java.awt.GraphicsEnvironment;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+
+import javax.imageio.ImageIO;
 
 public class CopyAreaOOB extends Canvas {
+    private static Frame frame;
+    private static Robot robot;
+    private static BufferedImage captureImg;
+    private static StringBuffer errorLog = new StringBuffer();
 
-    private static Robot robot = null;
+    public static void main(String[] args) throws Exception {
+        try {
+            robot = new Robot();
+            EventQueue.invokeAndWait(CopyAreaOOB::createTestUI);
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            EventQueue.invokeAndWait(() -> {
+                Point pt1 = frame.getLocationOnScreen();
+                Rectangle rect = new Rectangle(pt1.x, pt1.y, 400, 400);
+                captureImg = robot.createScreenCapture(rect);
```

Comment:
~~The captured image includes the frame decorations.~~ I see the frame is not decorated. Yet, the suggestion still applies.

You can use the location and the size of the canvas itself as the area to capture. For simplicity, save the canvas instance into a static variable.

I'd avoid hard-coding the width and height like this; constants would be serve the purpose, and if either width or height, or both, changes, all the code automatically updates to use the new values.

Reply:
Updated